### PR TITLE
Add `Verifier` for `MrSigner`

### PR DESCRIPTION
**Pull request by nick-mobilecoin**

### CODE REVIEW

**Comment by nick-mobilecoin** [comment]
Current dependencies on/for this PR:
* main
  * **PR #1** <a href="https://app.graphite.dev/github/pr/mobilecoinfoundation/attestation/1" target="_blank">Graphite</a> 
    * **PR #2** <a href="https://app.graphite.dev/github/pr/mobilecoinfoundation/attestation/2" target="_blank">Graphite</a> 
      * **PR #3** <a href="https://app.graphite.dev/github/pr/mobilecoinfoundation/attestation/3" target="_blank">Graphite</a> 
        * **PR #9** <a href="https://app.graphite.dev/github/pr/mobilecoinfoundation/attestation/9" target="_blank">Graphite</a> 
          * **PR #10** <a href="https://app.graphite.dev/github/pr/mobilecoinfoundation/attestation/10" target="_blank">Graphite</a> 
            * **PR #11** <a href="https://app.graphite.dev/github/pr/mobilecoinfoundation/attestation/11" target="_blank">Graphite</a> 
              * **PR #12** <a href="https://app.graphite.dev/github/pr/mobilecoinfoundation/attestation/12" target="_blank">Graphite</a> 
              * **PR #18** <a href="https://app.graphite.dev/github/pr/mobilecoinfoundation/attestation/18" target="_blank">Graphite</a> 
                * **PR #20** <a href="https://app.graphite.dev/github/pr/mobilecoinfoundation/attestation/20" target="_blank">Graphite</a> 
                  * **PR #21** <a href="https://app.graphite.dev/github/pr/mobilecoinfoundation/attestation/21" target="_blank">Graphite</a> 
                    * **PR #22** <a href="https://app.graphite.dev/github/pr/mobilecoinfoundation/attestation/22" target="_blank">Graphite</a> 
                      * **PR #23** <a href="https://app.graphite.dev/github/pr/mobilecoinfoundation/attestation/23" target="_blank">Graphite</a> 
                        * **PR #24** <a href="https://app.graphite.dev/github/pr/mobilecoinfoundation/attestation/24" target="_blank">Graphite</a>  👈
    * **PR #4** <a href="https://app.graphite.dev/github/pr/mobilecoinfoundation/attestation/4" target="_blank">Graphite</a> 

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/mobilecoinfoundation/attestation/24?utm_source=stack-comment).

File: verifier/src/report_body.rs [hunk 1]
```diff
@@ -274,6 +274,47 @@ impl<T: Accessor<MrEnclave>> Verifier<T> for MrEnclaveVerifier {
...
+    }
+}
+
+impl Accessor<MrSigner> for ReportBody {
+    fn get(&self) -> MrSigner {
+        let Measurement::MrSigner(mr_signer) = self.mr_signer() else {
+            panic!("`mr_signer()` should return a Measurement::MrSigner");
+        };
+        mr_signer
+    }
```
Comment (nick-mobilecoin) [inline verifier/src/report_body.rs:299]: Similar to #23
This is here until, https://github.com/mobilecoinfoundation/sgx/issues/286.
Added as a task to #7

File: verifier/src/report_body.rs [hunk 2]
```diff
@@ -29,7 +29,7 @@ pub trait Accessor<T>: Debug {
 
 /// [`Accessor`] for returning Self, i.e. T -> T
 macro_rules! self_accessor {
-    ($($type:ty)*) => {$(
+    ($($type:ty,)*) => {$(
```
Comment (varsha888) [inline verifier/src/report_body.rs:32]: What does this line do?

Comment (varsha888) [reply verifier/src/report_body.rs]: got it, thanks!

**Review by jcape — APPROVED** [review]
LGTM, but we'll eventually want to make this private to this crate, and replace it with a public API that checks versions in order to prevent misuse (i.e. people will absolutely say "I trust any enclave signed by this MRSIGNER, even if it's the wrong enclave or an ancient version that shouldn't work anymore"), e.g.:

```rust
pub struct MrSignerVerifier(And<InnerMrSignerVerifier, And<ExtProdId, IsvSvn>>);
```

**Comment by nick-mobilecoin** [comment]
Added MrSigner + Product ID to list on #7

**Review by varsha888 — APPROVED** [review]

**Pull request merged**